### PR TITLE
vstart: remove filestore options and config snippets

### PR DESCRIPTION
**Pull request by Svelar**
Remove filestore section from vstart.sh.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by Svelar** [comment]
jenkins retest this please

**Comment by Svelar** [comment]
jenkins retest this please

**Comment by Svelar** [comment]
jenkins retest this please

**Comment by Svelar** [comment]
jenkins retest this please

File: src/vstart.sh [hunk 1]
```diff
@@ -160,11 +160,7 @@ overwrite_conf=0
 cephx=1 #turn cephx on by default
 gssapi_authx=0
 cache=""
-if [ `uname` = FreeBSD ]; then
-    objectstore="filestore"
-else
-    objectstore="bluestore"
-fi
+objectstore="bluestore"
```
Comment (NitzanMordhai) [inline src/vstart.sh:163]: I recall FreeBDS doesn't support bluestore, maybe adding message in that case?

Comment (Svelar) [reply src/vstart.sh:163]: It means if using FreeBDS, error occurred, right?

Comment (idryomov) [reply src/vstart.sh:163]: @wjwithagen Can you chime in?  With filestore dropped in Reef, is there an ObjectStore backend can be used on FreeBSD?  If bluestore just won't work, perhaps we could default to memstore here for FreeBSD?

Comment (Svelar) [reply src/vstart.sh]: I don't have environment to test FreeBSD, but I have modified it to memstore, is it ok?

Comment (idryomov) [reply src/vstart.sh]: I don't have such an environment either.  It would be good to hear from @wjwithagen, or perhaps @trociny knows?

Comment (Svelar) [reply src/vstart.sh]: OK, wait for response, thank you so much.

Comment (trociny) [reply src/vstart.sh]: I have not tried Ceph on FreeBSD for long time, and I am sure @wjwithagen would know better, but AFAIR the bluestore is not supported on FreeBSD due to libaio dependency. I doubt it has changed recently, which means that Reef (server) will not be available for FreeBSD. 

I have no idea if memstore is supported on FreeBSD.

Taking that this vstart issue would be a highly minor problem for FreeBSD comparing with others I would not do anything FreeBSD specific here and leave it for people who want to work on making FreeBSD work.

Comment (idryomov) [reply src/vstart.sh]: > Taking that this vstart issue would be a highly minor problem for FreeBSD comparing with others I would not do anything FreeBSD specific here and leave it for people who want to work on making FreeBSD work.

Well, we already have a FreeBSD-specific branch in `vstart.sh` and need to do something about it...  If @wjwithagen doesn't chime in the course of a few days, I'd support a not-tested switch to memstore.

Comment (idryomov) [reply src/vstart.sh]: Alternatively, we could simply remove the `if [ `uname` = FreeBSD ]; then` statement...

Comment (Svelar) [reply src/vstart.sh]: Okay, I agree with you.

**Comment by Svelar** [comment]
jenkins test api

File: src/vstart.sh (hunk 1 above)
Comment (Svelar) [reply src/vstart.sh]: Hi @idryomov , is it time to merge this first? I prefer keeping FreeBSD's objectstore as memstore, if it goes wrong, someone could raise a PR then.

**Comment by Svelar** [comment]
jenkins test windows

**Review by idryomov — APPROVED** [review]

**Comment by idryomov** [comment]
> @idryomov Thanks, can I cherry pick from commits then raise a backport PR to reef, or should I open a tracker first?

No need for a ticket, just open a backport PR.

**Pull request merged**